### ISSUE TITLE
[Feature Request]: Add support for resize option with keeping aspect ratio and center crop for PrePostProcessor (Execution Graph)

### ISSUE THREAD

**Issue by ma7555** [openvinotoolkit/openvino#21088]
### Request Description

Some models require dynamic resizing of the input image along with a center crop. This feature does not seem to be supported and it is impossible to create this behaviour using `ov.preprocess.PrePostProcessor`

Example of PyTorch transforms:
```
transform_pt = transforms.Compose([
 transforms.Resize(256),
 transforms.CenterCrop(224)]
 )
```

[Resize](https://pytorch.org/vision/main/generated/torchvision.transforms.Resize.html): Takes the desired output size as an input. If size is a sequence like (h, w), output size will be matched to this. If size is an int, smaller edge of the image will be matched to this number. i.e, if height > width, then image will be rescaled to (size * height / width, size).

[CenterCrop](https://pytorch.org/vision/main/generated/torchvision.transforms.CenterCrop.html): Crops the given image at the center. If the image is torch Tensor, it is expected to have […, H, W] shape, where … means an arbitrary number of leading dimensions. If image size is smaller than output size along any edge, image is padded with 0 and then center cropped.

Is there a way to achieve the same behaviour with the current code base?

### Feature Use Case

It allows easier migration from PyTorch to Openvino IR

### Issue submission checklist

- [X] The feature request or improvement must be related to OpenVINO

**Comment by p-wysocki** [comment]
Hi @ma7555! Would our torchvision to OpenVINO preprocessing converter solve your issue? Both `Resize` and `CenterCrop` are supported. You can find the docs for it [here](https://github.com/openvinotoolkit/openvino/pull/20770), waiting to be merged.

**Comment by ma7555** [comment]
Hello @p-wysocki, It would work yes but as far as I understand it will add torchvision as a dependancy for the project. Am I correct?

Is there a solution for this before the new release?

**Comment by p-wysocki** [comment]
I believe the preprocessor converter should take `transforms.Compose()` result object as well, but that has to be checked - good question, I haven't through about that. That would eliminate the need for torchvision, if `transforms.Compose()` can be taken from torch itself.

> Is there a solution for this before the new release?

The torchvision to OpenVINO preprocessing converter is [already in the latest release](https://github.com/openvinotoolkit/openvino/tree/2023.2.0/src/bindings/python/src/openvino/preprocess/torchvision) - it's only the docs which are being reviewed and merged now. :)

You can find example usage in our [tests](https://github.com/openvinotoolkit/openvino/blob/2023.2.0/src/bindings/python/tests/test_torchvision_to_ov/test_preprocessor.py).

**Comment by p-wysocki** [comment]
Preprocess converter has been merged, you can view it [here](https://github.com/openvinotoolkit/openvino/tree/master/src/bindings/python/src/openvino/preprocess) more conveniently. Also:

> Is there a way to achieve the same behaviour with the current code base?

> This feature does not seem to be supported and it is impossible to create this behaviour using ov.preprocess.PrePostProcessor

If that's specifically what you're after, the torch to OpenVINO preprocessing converter has been implemented fully using OpenVINO PrePostProcessor. You can take a look at how we did it and draw inspiration from that, if the current converter is not satisfactory:
- [CenterCrop](https://github.com/openvinotoolkit/openvino/blob/2023.2.0/src/bindings/python/src/openvino/preprocess/torchvision/torchvision_preprocessing.py#L257)
- [Resize](https://github.com/openvinotoolkit/openvino/blob/2023.2.0/src/bindings/python/src/openvino/preprocess/torchvision/torchvision_preprocessing.py#L257)

If you decide to implement the preprocessing functions on your own please let us know - we would love to improve our code.

**Comment by ma7555** [comment]
Hello @p-wysocki, I have tested the implementation of these methods however, they returned different results from the PyTorch transforms. I have aslo found out that the only accepted resizing method is '[NEAREST](https://github.com/openvinotoolkit/openvino/blob/cfd42bd2cb033929dd1858df3760bf23d967adac/src/bindings/python/src/openvino/preprocess/torchvision/torchvision_preprocessing.py#L290)'.

What works and is identical:

1. ToTensor
2. Normalize

What is not working (gives different result):

1. Resize (even NEAREST)
2. CenterCrop

```
import torch
from torchvision import transforms
from PIL import Image
import numpy as np

img = Image.open('/content/IMG_1311.jpg')

transform2 = transforms.Compose([
transforms.Resize(256, transforms.InterpolationMode.NEAREST),
 transforms.CenterCrop(224),
 transforms.ToTensor(),
 transforms.Normalize(
  mean=[0.485, 0.456, 0.406],
  std=[0.229, 0.224, 0.225]
 )
 ]
 )

device = "cpu"
model_path = "/content/models/feature_extract_compressed_entire_image_15k_fine_tune_model.pth"
torch_model = torch.load(model_path, map_location=device)
torch_model.eval()

torch_input = transform2(img)
torch_input = torch.unsqueeze(torch_input, 0)
batch_t = torch_input.to(device)
out = torch_model(batch_t)
with torch.no_grad():
    outputs = torch.softmax(out, dim=-1)
    preds = outputs.to(device).numpy()
    live_prob = preds[:, 1]
print(live_prob)
```
```
[0.69592786]
```

```
import openvino as ov
import scipy
import numpy as np
from openvino.preprocess import ResizeAlgorithm, ColorFormat
from openvino.runtime import Type
from openvino.preprocess.torchvision import PreprocessConverter

img = Image.open('/content/IMG_1311.jpg')

ov_model = ov.convert_model(torch_model, example_input=torch_input)

ov_model = PreprocessConverter.from_torchvision(
        model=ov_model, transform=transform2, input_example=img,
    )
core = ov.Core()
img_ov = np.expand_dims(np.array(img), 0)
compiled_model = core.compile_model(ov_model, "CPU")
x = compiled_model(img_ov)
scipy.special.softmax(x[0])[:, 1]
```
```
[0.97624683]
```

**Comment by ma7555** [comment]
I guess the problem with the Resize method is that when Resize is given only one dimenstion Resize(256) it resizes while keeping the aspect ratio intact. It only resizes the shorter dimension to 256 while openvino always expects two dimensions resizing. (256, 256) as an example.

**Comment by p-wysocki** [comment]
Good catch @ma7555, I'll open a ticket for that. I believe our tests missed it since both tests using Resize are using square images. Thanks! Until the ticket is closed, this can be worked around by writing a custom preprocessing step using OpenVINO PrePostProcessor.

**Comment by p-wysocki** [comment]
Issue tracked in an internal ticket 125609
